ikura/shim/src/cli.rs

```diff
@@ -120,19 +108,67 @@ pub enum Commands {
 pub mod serve {
     //! CLI definition for the `serve` subcommand.
 
-    use super::{DockParams, IkuraRpcParams, KeyManagementParams};
-    use clap::Args;
+    use super::{DockParams, IkuraRpcParams, KeyManagementParams, ENV_IKURA_NAMESPACE};
+    use clap::{Args, Subcommand};
 
     #[derive(Debug, Args)]
     pub struct Params {
-        #[clap(flatten)]
-        pub rpc: IkuraRpcParams,
+        #[command(subcommand)]
+        pub dock: Dock,
+    }
+
+    #[derive(Subcommand, Debug)]
+    pub enum Dock {
```

Comment:
Just noting: the biggest implication of this change, it seems, is that you can no longer run the shim for _all_ docks at the same time. I am not sure why that would be useful & don't think this fundamentally impacts the product. I'm sure we could add it again later if needed, anyway.